src/Controllers/ImageController.php

```diff
@@ -72,6 +72,65 @@ public function __invoke(Request $request, int $store, string $size, string $pla
         return $this->storage()->response($resizedPath);
     }
 
+    public function getResizedPath(string $size, string $file, string $webp)
+    {
+        foreach (config('imageresizer.external') as $placeholder => $url) {
+            if (Str::startsWith($file, $placeholder)) {
+                $file = Str::replaceFirst($placeholder, '', $file);
+                $placeholderUrl = $url;
+                break;
+            }
+        }
+
+        $placeholder = isset($placeholderUrl)
+            ? $placeholder
+            : 'local';
+
+        return config('rapidez.store').'/resizes/'.$placeholder.'/'.$size.'/'.$file.$webp;
+    }
+
+    public function redirectFromSku(Request $request, string $size, string $file)
+    {
+        $webp = str_ends_with($file, '.webp') ? '.webp' : '';
+        $baseFile = $webp ? str_replace_last('.webp', '', $file) : $file;
+
+        $sku = pathinfo($baseFile)['filename'];
+        $image = $this->productImageUrlFromSku($sku);
+
+        if ($image == 'magento/catalog/placeholder.jpg') {
+            return redirect($this->getResizedPath($size, $image, $webp));
+        }
+
+        $pathSku = $this->getResizedPath($size, 'magento/sku/'.$baseFile, $webp);
+        $pathImage = $this->getResizedPath($size, $image, $webp);
+
+        if (!$this->storage()->directoryExists(dirname($pathSku))) {
+            $this->storage()->createDirectory(dirname($pathSku));
+        }
+
+        if ($this->storage()->has($pathImage)) {
+            if (is_link($this->storage()->path($pathSku))) {
+                unlink($this->storage()->path($pathSku));
+            }
+            symlink($this->storage()->path($pathImage), $this->storage()->path($pathSku));
```

Comment:
Flysystem doens't support symlinks? What if the storage: https://github.com/rapidez/image-resizer/blob/master/config/imageresizer.php#L7 is switched to a S3 storage?

Reply:
Hmm, I guess that's a problem with S3 (in hindsight, understandably) not supporting symlinks, I'd kinda thought it was universal. We could just make it store the image file multiple times in this case but I feel like that shouldn't be necessary.

Alternatively, we might be able to check for symlink support, then skip the symlink and jump straight to a permanent redirect for those cases?

@indykoning

Reply:
It's not even necessarily S3 that doesn't support symlinks. It is flysystem and they do it on purpose https://github.com/thephpleague/flysystem/issues/474 for security and because not every service supports symlinks (Like S3, Dropbox, Drive)

You could check if the current storage is an instanceof local and use symlinks, otherwise redirect. But be careful of
https://github.com/thephpleague/flysystem/blob/3.x/src/SymbolicLinkEncountered.php
As symlink handling is disabled by default 
https://github.com/thephpleague/flysystem/blob/b4d2b9d820cce156ed8daa997d467e290aafb9a8/src/Local/LocalFilesystemAdapter.php#L76
which may cause errors when trying to access images via symlinks
https://github.com/thephpleague/flysystem/blob/b4d2b9d820cce156ed8daa997d467e290aafb9a8/src/Local/LocalFilesystemAdapter.php#L216

Plus with all these checks we may just be bloating the code.
In the end what is faster, and more efficient?

Redirecting from one url to another. Where the server and your browser only caches one instance of the image.
Or symlinking the image. Where you save on a request, and a potential start of the Framework?

I don't know. I havent tested the options in a production setting.

Reply:
I've reverted it to the permanent redirect version for now. It will definitely be worth looking into optimizing this in the future as it is a significant performance difference between directly serving the file vs having to start up the framework, but I don't really know what the approach should be for that.

Reply:
Here a little explanation how 301 caching works on nginx
http://nginx.org/en/docs/http/ngx_http_proxy_module.html#proxy_cache_valid
so we could cache all 301s

OR better
> If the header does not include the “X-Accel-Expires” field, parameters of caching may be set in the header fields “Expires” or “Cache-Control”

So possibly adding `->setPublic()->setMaxAge(3600)` after the redirect should allow it to cache the redirect for an hour on most backends, preventing Laravel from having to be booted again